public/app/features/alerting/unified/hooks/useCombinedRuleNamespaces.ts

```diff
@@ -34,39 +40,51 @@ export function useCombinedRuleNamespaces(): CombinedRuleNamespace[] {
 
         // first get all the ruler rules in
         Object.entries(rulerRules || {}).forEach(([namespaceName, groups]) => {
-          namespaces[namespaceName] = {
+          const namespace: CombinedRuleNamespace = {
             rulesSource,
             name: namespaceName,
-            groups: groups.map((group) => ({
-              name: group.name,
-              rules: group.rules.map(
-                (rule): CombinedRule =>
-                  isAlertingRulerRule(rule)
-                    ? {
-                        name: rule.alert,
-                        query: rule.expr,
-                        labels: rule.labels || {},
-                        annotations: rule.annotations || {},
-                        rulerRule: rule,
-                      }
-                    : isRecordingRulerRule(rule)
-                    ? {
-                        name: rule.record,
-                        query: rule.expr,
-                        labels: rule.labels || {},
-                        annotations: {},
-                        rulerRule: rule,
-                      }
-                    : {
-                        name: rule.grafana_alert.title,
-                        query: '',
-                        labels: rule.grafana_alert.labels || {},
-                        annotations: rule.grafana_alert.annotations || {},
-                        rulerRule: rule,
-                      }
-              ),
-            })),
+            groups: [],
           };
+          namespaces[namespaceName] = namespace;
+          namespace.groups = groups.map((group) => {
+            const combinedGroup: CombinedRuleGroup = {
+              name: group.name,
+              rules: [],
+            };
+            combinedGroup.rules = group.rules.map(
+              (rule): CombinedRule =>
```

Comment:
nit: I think this code should be split up into multiple smaller functions to improve readability and prevent nesting.

Reply:
- [ ] break it up a little

Reply:
broke it up into several hopefully well-named functions